gatsby-node.js

```diff
@@ -5,3 +5,53 @@
  */
 
 // You can delete this file if you're not using it
+
+const path = require('path');
+const { createFilePath } = require('gatsby-source-filesystem');
+
+exports.createPages = ({ actions, graphql}) => {
+  const { createPage } = actions;
+
+  const postTemplate = path.resolve('src/templates/postTemplate.js');
+
+  return graphql(`
+    {
+      allMarkdownRemark(
+        sort: { order: DESC, fields: [frontmatter___date] }
+        limit: 1000
+      ) {
+        edges {
+          node {
+            fields {
+              slug
+            }
+          }
+        }
+      }
+    }
+  `).then(result => {
+    if (result.errors) {
+      return Promise.reject(result.errors);
+    }
+
+    result.data.allMarkdownRemark.edges.forEach(({ node }) => {
+      createPage({
+        path: node.fields.slug,
+        component: postTemplate,
+        context: { slug: node.fields.slug },
+      });
+    });
+  });
+};
+
+exports.onCreateNode = ({ node, actions, getNode }) => {
```

Comment:
Creates the pages automatically based on a slug rather than relying on the title manually being entered.